gm4_block_compressors/data/block_compressors/functions/process.mcfunction

```diff
@@ -0,0 +1,3 @@
+execute positioned ~ ~1 ~ as @e[type=item,distance=..1] positioned ~ ~-1 ~ run function block_compressors:item
+particle portal ~ ~1.1 ~ 0 0 0 0.1 20
+data merge entity @s {Fire:2000}
```

Comment:
I still think you're missing a destroy function to return the compressor's items to the player and remove the armor stand when the machine is broken.

Reply:
So how does this work? In theory the custom crafter main function could run before the block compressor and remove the entity before the block compressor main function has a chance to spawn the extra items.